File: DSA-Python/2-patterns/17-pattern.py
'''
    A
   ABA
  ABCBA
 ABCDCBA
ABCDEDCBA
'''


def main(n:int) -> None:
    for i in range(1, n+1):
        for j in range(n-i, 0, -1):
            print(end=' ')
        
        for j in range(65, 65+i):
            print(chr(j), end='')
            
        for j in range(1, i):
            print(chr(i-j+64), end='')
        print()
    
    
if __name__ == "__main__": 
    n = int(input())
    main(n)

File: DSA-C++/3-patterns/17-pattern.cpp
#include <bits/stdc++.h>

using namespace std;

//     A
//    ABA
//   ABCBA
//  ABCDCBA
// ABCDEDCBA


int main() {
    int n;
    cin >> n;

    for (int i=1; i<=n; ++i) {
        int c = 64;

        for (int j=1; j<=n-i; ++j) {
            cout << " ";
        }

        for (int j=n-i+1; j<=n; ++j) {
            cout << static_cast<char>(++c);
        }

        for (int j=1; j<i; ++j) {
            cout << static_cast<char>(--c);
        }
        cout << endl;
    }
    
    return 0;
}